src/test/java/seedu/address/storage/JsonSerializableResumeBookTest.java

```diff
@@ -0,0 +1,168 @@
+
+    private static final Path INVALID_SKILL_FILE = TEST_DATA_FOLDER.resolve("invalidSkillResumeBook.json");
+    private static final Path DUPLICATE_SKILL_FILE = TEST_DATA_FOLDER.resolve("duplicateSkillResumeBook.json");
+
+    @Test
+    public void toModelType_typicalFile_success() throws Exception {
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(TYPICAL_RESUME_BOOK_FILE,
+                JsonSerializableResumeBook.class).get();
+        ResumeBook resumeBookFromFile = dataFromFile.toModelType();
+        ResumeBook typicalResumeBook = TypicalResumeBook.TYPICAL_WITHOUT_GOOGLE;
+        assertEquals(resumeBookFromFile, typicalResumeBook);
+    }
+
+    @Test
+    public void toModelType_invalidResumeFile_throwsIllegalValueException() throws Exception {
+        // The file contains an invalid resume name
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(INVALID_RESUME_FILE,
+                JsonSerializableResumeBook.class).get();
+        assertThrows(IllegalValueException.class, Name.MESSAGE_CONSTRAINTS, dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_duplicateResumeFile_throwsIllegalValueException() throws Exception {
+        // The file contains duplicates of resume which have a same name
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(DUPLICATE_RESUME_FILE,
+                JsonSerializableResumeBook.class).get();
+        assertThrows(IllegalValueException.class, MESSAGE_DUPLICATE_RESUME, dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidSkillFile_throwsIllegalValueException() throws Exception {
+        // The file contains an invalid skill with invalid level
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(INVALID_SKILL_FILE,
+                JsonSerializableResumeBook.class).get();
+        assertThrows(IllegalValueException.class, Level.MESSAGE_CONSTRAINTS, dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_duplicateSkillFile_throwsIllegalValueException() throws Exception {
+        // The file contains duplicates of skill which have a same name
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(DUPLICATE_SKILL_FILE,
+                JsonSerializableResumeBook.class).get();
+        assertThrows(IllegalValueException.class, MESSAGE_DUPLICATE_SKILL, dataFromFile::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidInternshipFile_throwsIllegalValueException() throws Exception {
+        // The file contains an invalid internship wherein "from" is after "to"
+        JsonSerializableResumeBook dataFromFile = JsonUtil.readJsonFile(INVALID_INTERNSHIP_FILE,
+                JsonSerializableResumeBook.class).get();
+        assertThrows(
+                IllegalValueException.class,
+                "The \"to\" field must not precede the \"from\" field.", dataFromFile::toModelType);
```

Comment:
I think Hai changed this to 'to' and 'from' instead of "to" and "from". I can't remember which PR. @helloImHai can confirm?

Reply:
It was more about the word precede / after bc it can imply where it is in the command and not the time itself, Duong suggested to use "'from' cannot be later than 'to' field. "